### PR TITLE
Add types to Type

### PR DESCRIPTION
**Pull request by spawnia**

### CODE REVIEW

File: src/Type/Schema.php [hunk 1]
```diff
@@ -410,11 +410,16 @@ private function defaultTypeLoader(string $typeName): ?Type
...
+     * @param Type|callable $type
+     *
+     * @template T of Type
+     * @phpstan-param T|callable():T $type
+     * @phpstan-return T
      */
     public static function resolveType($type): Type
     {
         if ($type instanceof Type) {
+            /** @phpstan-var T $type */
```
Comment (simPod) [inline src/Type/Schema.php:422]: I'd expect phpstan to remember the type information from 

```php
* @param Type|callable $type
* @phpstan-param T|callable():T $type
```

🤔

File: src/Type/Definition/NonNull.php [hunk 2]
```diff
@@ -26,16 +26,20 @@ public function toString(): string
...
+
+        return $type;
     }
 
-    /**
-     * @return (NullableType&Type)
-     */
     public function getWrappedType(bool $recurse = false): Type
     {
+        /** @var NullableType&Type $type */
```
Comment (simPod) [inline src/Type/Definition/NonNull.php:42]: is this needed? It's typed above

image

File: tests/Executor/ExecutorTest.php [hunk 3]
```diff
@@ -54,7 +54,7 @@ public function testExecutesArbitraryCode(): void
             });
         };
 
-        $data = [
+        $data     = [
```
Comment (simPod) [inline tests/Executor/ExecutorTest.php:57]: wasn't it compliant with CS already before?

**Review by simPod — APPROVED** [review]

File: src/Type/Schema.php (hunk 1 above)
Comment (spawnia) [reply src/Type/Schema.php:422]: Me too, it seems as though the `instanceof` check erases the generic type. I still think it is better to have the function be well-typed for consumers.

File: src/Type/Definition/NonNull.php (hunk 2 above)
Comment (spawnia) [reply src/Type/Definition/NonNull.php:42]: I would rather omit it, PHPStan will not let me.

File: tests/Executor/ExecutorTest.php (hunk 3 above)
Comment (spawnia) [reply tests/Executor/ExecutorTest.php:57]: I tried reverting, seems like the newest version changes this. This is one of the reasons why I do not like alignment, my preference would be to left-align everything always.

Comment (simPod) [reply tests/Executor/ExecutorTest.php:57]: After some time, I agree. There's even no proper IDE support for this kind of alignment. It some cases it's really handy but overall not worth I guess.

File: src/Type/Definition/NonNull.php (hunk 2 above)
Comment (simPod) [reply src/Type/Definition/NonNull.php:42]: Aight, let's skip this for now.

Eg. psalm can detect useless `@var` annotations so we can automatically clean this up later

File: tests/Executor/ExecutorTest.php (hunk 3 above)
Comment (spawnia) [reply tests/Executor/ExecutorTest.php:57]: As far as I can tell, PHP_CodeSniffer has no option to force left-alignment. Wading through their docs is a total pain in the neck though, perhaps we should go with php-cs-fixer.

**Pull request merged**